test/compilable/fail11720.d

```diff
@@ -0,0 +1,23 @@
+void main()
+{
+    foo();
```

Comment:
This test should create multiple different instantiations of `foo` given https://github.com/dlang/dmd/pull/11250#discussion_r437891414

Reply:
I don't follow you, it's only called once.

Reply:
can you expand the problem ?

Reply:
The problem I meant is that there are only 2 statics generated in foo, one generating the original symbol, one with the aliased symbol. A third static would verify that the generated symbols don't collide. IIRC the version I commented on didn't use `generateId` so it might be solved now to some extend, but a test (a third static generated at the same location) could verify that.

There is still a problem with `generateId`, though: with separate compilation units, the same symbol name can be generated, but might have different implementations.

Reply:
Ok thanks for the explanations. Actually I'm embarassed because this was not supposed to be covered by the PR but it had appeared that the test passed.